x-pack/plugins/data_visualizer/public/application/file_data_visualizer/components/file_contents/field_badge.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC } from 'react';
+import { EuiBadge, EuiFlexGroup, EuiFlexItem, EuiToolTip } from '@elastic/eui';
+import { FieldIcon } from '@kbn/react-field';
+import { i18n } from '@kbn/i18n';
+import { getSupportedFieldType } from '../../../common/components/fields_stats_grid/get_field_names';
+import { useCurrentEuiTheme } from '../../../common/hooks/use_current_eui_theme';
+
+interface Props {
+  type: string | undefined;
+  value: string;
+  name: string;
+}
+
+export const FieldBadge: FC<Props> = ({ type, value, name }) => {
+  const { euiColorLightestShade, euiColorLightShade } = useCurrentEuiTheme();
+  const supportedType = getSupportedFieldType(type ?? 'unknown');
+  const tooltip = type
+    ? i18n.translate('xpack.dataVisualizer.file.fileContents.fieldBadge.tooltip', {
+        defaultMessage: 'Type: {type}',
+        values: { type: supportedType },
+      })
+    : undefined;
+  return (
+    <EuiToolTip title={name} content={tooltip}>
+      <EuiBadge
+        css={{
+          marginRight: '2px',
```

Comment:
I experimented with using Eli sizes here but I didn't feel they added much value and actually made it harder to understand.
Also, only `4px` could be substituted for an eui variable.
Unfortunately these magic numbers are necessary to align the badges up correctly with the text.